src/scripts/main.js

```diff
@@ -1,3 +1,223 @@
 'use strict';
 
-// write code here
+const mainTable = document.querySelector('table');
+const theadClickArea = mainTable.querySelector('thead');
+const bodyTable = mainTable.querySelector('tbody');
+
+let selectFlag = false;
+
+//  Create a function for storing state of search button(th),
+//  it helps calculate function userSort().
+function renewDataFlag() {
+  let prevSelect = '';
+  let flagState = false;
+
+  return function(currentElem) {
+    if (currentElem.innerText === prevSelect) {
+      flagState = !flagState;
+
+      return flagState;
+    }
+    flagState = false;
+    prevSelect = currentElem.innerText;
+
+    return flagState;
+  };
+}
+
+// Making closure for base state of search button(th).
+const checkFlag = renewDataFlag();
+
+function userSort() {
+  let sortFlag = false;
+
+  theadClickArea.addEventListener('click', (e) => {
+    const mainTH = e.target.closest('th');
+    const indexCol = mainTH.cellIndex;
+    const allRows = [...bodyTable.children];
+
+    if (!mainTH) {
+      return;
+    }
+    sortFlag = checkFlag(mainTH);
+
+    allRows.sort((firstRow, secondRow) => {
+      let argFirst = firstRow.cells[indexCol].innerText;
+      let argSecond = secondRow.cells[indexCol].innerText;
+
+      if (mainTH.innerText === 'Salary') {
+        argFirst = +argFirst.match(/\d/g).join('');
+        argSecond = +argSecond.match(/\d/g).join('');
+
+        return sortFlag
+          ? argSecond - argFirst
+          : argFirst - argSecond;
+      } else {
+        return sortFlag
+          ? argSecond.localeCompare(argFirst)
+          : argFirst.localeCompare(argSecond);
+      }
+    });
+
+    allRows.map(row => bodyTable.append(row));
+  });
+}
+
+userSort(mainTable);
+
+bodyTable.addEventListener('click', (e) => {
+  const item = e.target.closest('tr');
+
+  if (!selectFlag) {
+    selectFlag = true;
+    item.classList.toggle('active');
+
+    return;
+  }
+
+  document.querySelector('.active').classList.remove('active');
+  item.classList.toggle('active');
+});
+
+bodyTable.addEventListener('dblclick', (e) => {
+  const input = document.createElement('input');
+  const selectedItem = e.target.closest('td');
+  const innerText = selectedItem.innerHTML;
+
+  input.classList.add('cell-input');
+  input.setAttribute('size', '1');
+  selectedItem.innerHTML = '';
+  selectedItem.append(input);
+
+  input.focus();
+
+  const eventKeys = (event1) => {
+    if (event1.keyCode === 13) {
+      if (input.value) {
+        selectedItem.innerHTML = input.value;
+      } else {
+        selectedItem.innerHTML = innerText;
+      }
+    }
+  };
+
+  const eventBlur = (event2) => {
+    event2.stopImmediatePropagation();
+
+    if (input.value) {
+      selectedItem.innerHTML = input.value;
+    } else {
+      selectedItem.innerHTML = innerText;
+    }
+  };
+
+  input.addEventListener('keypress', eventKeys, true);
+  input.addEventListener('blur', eventBlur, false);
```

Comment:
having 2 different callbacks for different event listeners is okay (what we discussed on a q&a yesterday)